File: absolute-zero/examples/fortran/nop.f90
! Certified Null Operation in Fortran
!
! A program that does absolutely nothing at the application level.
! Exits with code 0 (success) without any observable side effects.
!
! Properties:
! - Terminates immediately
! - No I/O operations
! - No array operations
! - No mathematical computation
! - Exit code 0
!
! Compile: gfortran nop.f90 -o nop
! Run: ./nop
!
! Historical context:
! FORTRAN (FORmula TRANslation) was developed by John Backus and his
! team at IBM, with the first compiler released in 1957. It was the
! first high-level programming language and revolutionized scientific
! and engineering computing.
!
! FORTRAN's historical significance:
! - First widely adopted high-level language
! - Introduced the concept of optimizing compilers
! - Made computers accessible to scientists and engineers
! - Established conventions still used today (array indexing, DO loops)
! - Enabled numerical computation at unprecedented scales
!
! Fortran (modern lowercase spelling) continues to evolve:
! - Fortran 90: Major modernization, free-form source
! - Fortran 95: Minor refinements
! - Fortran 2003: Object-oriented features, C interoperability
! - Fortran 2008: Coarrays for parallel programming
! - Fortran 2018: Further parallel and team features
!
! This CNO uses Fortran 90+ free-form syntax.

program nop
    implicit none

    ! Empty program body - the minimal CNO in Fortran
    ! No computation, no I/O, immediate termination

end program nop

! Verification notes:
! - Fortran runtime initializes I/O units
! - Implicit type system is established (but not used here)
! - At application level: CNO
! - At runtime level: initialization of standard units (5,6,etc.)
!
! The 'implicit none' statement is modern Fortran best practice,
! requiring all variables to be explicitly declared. Here, we have
! no variables, so it's vacuously satisfied.
!
! Legacy notes:
! - Much of the world's scientific computing infrastructure uses Fortran
! - Numerical libraries (LAPACK, BLAS, etc.) are primarily Fortran
! - Fortran remains the fastest language for many numerical tasks
! - Modern Fortran is significantly different from FORTRAN 77
! - Array operations and parallelism are first-class features
!
! Fixed-form (FORTRAN 77) equivalent:
!       PROGRAM NOP
!       END
!
! The evolution from punch-card columns to free-form syntax
! represents Fortran's journey from the 1950s to modern computing.
!
! This CNO demonstrates the minimal program structure while adhering
! to modern Fortran conventions. Even with no computation, the program
! structure reflects decades of language evolution and best practices.

File: absolute-zero/examples/swift/Nop.swift
/**
 * Certified Null Operation in Swift
 *
 * A program that does absolutely nothing at the application level.
 * Exits with code 0 (success) without any observable side effects.
 *
 * Properties:
 * - Terminates immediately
 * - No I/O operations
 * - No memory allocation beyond runtime overhead
 * - Exit code 0
 *
 * Compile: swiftc Nop.swift -o nop
 * Run: ./nop
 */

// Empty file - the minimal CNO in Swift
// Swift allows top-level code, so an empty file is a valid program
// that does nothing and exits with code 0.

/*
 * Verification notes:
 * - Swift runtime initializes (ARC, standard library, etc.)
 * - No explicit main function required in scripts
 * - For compiled programs, an implicit main is generated
 * - At application level: CNO
 * - At runtime level: memory management initialization
 *
 * This demonstrates Swift's dual nature as both a scripting
 * and compiled systems language. An empty file is valid in both contexts.
 *
 * Alternative explicit form:
 *
 * @main
 * struct NopProgram {
 *     static func main() {
 *         // Empty main
 *     }
 * }
 *
 * Both forms are equally valid CNOs.
 */
